spec/support/shared_examples/ca.rb

```diff
@@ -0,0 +1,88 @@
+shared_examples "a CA" do
+
+  it { is_expected.to respond_to(:create) }
+  it { is_expected.to respond_to(:create_keypair) }
+  it { is_expected.to respond_to(:create_keypair_in) }
+  it { is_expected.to respond_to(:reference_keypair) }
+  it { is_expected.to respond_to(:<=>) }
+
+  before do
+    @ca = described_class.create("some-ca", "a-bucket")
```

Comment:
Not actually important but you might like to use an explicit subject here, to make it clear this is the object under test: https://relishapp.com/rspec/rspec-core/v/3-2/docs/subject/explicit-subject

Reply:
Also: if you `let` "some-ca" and "a-bucket" as `:ca` and `:bucket` they can be used in subsequent tests, to make these tests less brittle.